snippets/base-class.js

```diff
@@ -0,0 +1,82 @@
+/* We create a new class that extends from the native HTMLElement */
+class BatchingElement extends HTMLElement {
+  constructor() {
+
+    /* We now have to call `super` to make sure our element gets set up correctly */
+    super();
+    this.updateComplete = this.__createDeferredPromise();
+  }
+
+  async requestUpdate() {
+    if (!this.updateRequested) {
+      this.updateRequested = true;
+      this.updateRequested = await false;
+      this.update();
+      this.__resolve();
+      this.updateComplete = this.__createDeferredPromise();
+    }
+  }
+
+  update() {}
+
+  __createDeferredPromise() {
+    return new Promise((resolve) => {
+      this.__resolve = resolve;
+    });
+  }
+}
+
+/* We create a new class, MyElement, that extends from our BatchingElement base class */
+class MyElement extends BatchingElement {
+
+  constructor() {
+    super();
+    /* We attach a shadowRoot to our component for good measure */
+    this.attachShadow({ mode: "open" });
+  }
+
+  connectedCallback() {
+    /* And we call an initial render when our component gets connected to the DOM */
+    this.update();
+  }
+
+  /**
+   * This is our `update` function that will get triggered by the
+   * `requestUpdate` method.
+   *
+   * Any time we set a property on this element, we'll also
+   * trigger an update.
+   */
+  update() {
+    console.log('updating!');
+    this.shadowRoot.innerHTML = `
```

Comment:
even in this case I'd rather see `<template>` being used. You can target spans by ID or maybe imperatively append empty text nodes

Reply:
Hmm yeah... I see what you're saying. On the other hand I think maybe using a template/cloning/queryselecting will distract from whats important here (the async-ey stuff), it also adds quite a couple lines of code, where innerHTML is just one statement